src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -1606,9 +1606,13 @@ private void InheritNullableStateOfFieldOrProperty(int targetContainerSlot, int
             }
             else if (EmptyStructTypeCache.IsTrackableStructType(fieldOrPropertyType.TypeSymbol))
             {
-                InheritNullableStateOfTrackableStruct(fieldOrPropertyType.TypeSymbol, 
-                                                      GetOrCreateSlot(fieldOrProperty, targetContainerSlot),
-                                                      valueContainerSlot > 0 ? GetOrCreateSlot(fieldOrProperty, valueContainerSlot) : -1, isByRefTarget);
+                var slot = GetOrCreateSlot(fieldOrProperty, targetContainerSlot);
```

Comment:
This is how I read this: if the slot is -1 the struct is empty, so everything is at the top value. If the slot is 0... this is unreachable? So everything is at the top value. If the slot is greater than zero, the state is the value of the transition function from the previous state.

Is this right? If so, maybe we should do something to make this more readable. For instance, can this ever be 0? I would be surprised.